### ISSUE TITLE
Possible bug: color in `line` and `text` theme attributes not working as expected

### ISSUE THREAD

**Issue by mvanaman** [tidyverse/ggplot2#4947]
According to the help documentation for `?theme`, line attributes (e.g., `axis.line`) should inherit from `line`, text attributes (e.g., `axis.text`) should inherit from `text`, and so on. However, when trying to set the color of all lines with `line =`, colors do not change (the size does change as expected, though).

Similarly for `text`, when adding e.g., `theme(text = element_text(color = "green", size = 10))`, size affects both axis titles and axis text, but color only affects axis titles.

Am I correct that this behavior is unexpected?

``` r
library(ggplot2)
ggplot(iris, aes(x = Sepal.Width, Petal.Length)) +
  geom_point() +
  geom_smooth() +
  theme(line = element_line(color = "green", size = 10))
#> `geom_smooth()` using method = 'loess' and formula 'y ~ x'
```

![](https://i.imgur.com/5UI7gaM.png)

<sup>Created on 2022-08-15 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by karawoo** [comment]
This is a common source of confusion. The default theme, `theme_grey()`, sets some of these elements directly. Theme elements only inherit from other elements if they aren't explicitly set. This talk explains how it works in more detail: https://resources.rstudio.com/resources/rstudioglobal-2021/always-look-on-the-bright-side-of-plots/

**Comment by mvanaman** [comment]
I think I understand now. Interesting. Intuitively, one would expect that setting a higher order attribute would override all of the lower order attributes, even if they were set by the default team, since all the other layers are order–dependent. I'm sure there's a technical reason why this little slice of ggplot works differently, but I'm grateful to understand a little better now. 

TL; DR: thanks for your fast response and for the link to that very helpful video.
